### PR TITLE
Fix low-power HEVC encode

### PR DESCRIPTION
**Pull request by onabiull**
Fixes: #1805

### CODE REVIEW

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp [hunk 1]
```diff
@@ -1961,16 +1963,8 @@ mfxStatus CheckVideoParam(MfxVideoParam& par, MFX_ENCODE_CAPS_HEVC const & caps,
     if (par.mfx.TargetUsage && caps.ddi_caps.TUSupport)
         changed += CheckTU(caps.ddi_caps.TUSupport, par.mfx.TargetUsage);
 
-    changed += CheckMax(par.mfx.GopRefDist, caps.ddi_caps.SliceIPOnly ? 1 : (par.mfx.GopPicSize ? std::max(1, par.mfx.GopPicSize - 1) : 0xFFFF));
+    changed += CheckMax(par.mfx.GopRefDist, (caps.ddi_caps.SliceIPOnly || IsOn(par.mfx.LowPower)) ? 1 : (par.mfx.GopPicSize ? std::max(1, par.mfx.GopPicSize - 1) : 0xFFFF));
 
-#if (MFX_VERSION >= MFX_VERSION_NEXT)
```
Comment (saosipov) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: Why is this part removed?
RAB are not supported on TGL tu7

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp [hunk 2]
```diff
@@ -2220,23 +2214,6 @@ mfxStatus CheckVideoParam(MfxVideoParam& par, MFX_ENCODE_CAPS_HEVC const & caps,
 
     changed += CheckOption(par.mfx.NumSlice, MakeSlices(par, caps.ddi_caps.SliceStructure), 0);
 
-    if (par.m_ext.CO2.NumMbPerSlice != 0)
```
Comment (saosipov) [inline _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: Why is this part removed?

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp (hunk 1 above)
Comment (onabiull) [reply _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: RAB-related lines cause #1805. Can't exactly say why.

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp (hunk 2 above)
Comment (onabiull) [reply _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: fixed

File: _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp (hunk 1 above)
Comment (saosipov) [reply _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: It looks like it was our guy who fixed sample_encode + mss recently. I will check with him

Comment (onabiull) [reply _studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_hw_par.cpp]: Take a look at #1086 also. Looks like something missed here.
